pkg/controller/direct/logging/diffs.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
```

Comment:
This does not technically need to live here, right? We can move it out of the `logging` package as a follow on maybe? I'd also like to add some units to it.

Reply:
Let's leave it in logging, and once another controller needs this, we can better understand how to adjust this lib.